flink-streaming-java/src/main/java/org/apache/flink/streaming/util/HDFSCopyFromLocal.java

```diff
@@ -62,4 +64,34 @@ public void run() {
 			throw asyncException.f0;
 		}
 	}
+
+	/**
+	 * Ensure that target path terminates with a new directory to be created by fs. If remoteURI does not specify a new
+	 * directory, append local directory name.
+	 * @param fs
+	 * @param localPath
+	 * @param remoteURI
+	 * @return
+	 * @throws IOException
+	 */
+	protected static URI checkInitialDirectory(final FileSystem fs,final File localPath, final URI remoteURI) throws IOException {
+		if (localPath.isDirectory()) {
+			Path remotePath = new Path(remoteURI);
+			if (fs.exists(remotePath)) {
+				return new Path(remotePath,localPath.getName()).toUri();
+			}
+		}
+		return remoteURI;
+	}
+
+	protected static void copyFromLocalFile(final FileSystem fs, final File localPath, final URI remotePath) throws Exception {
```

Comment:
Sorry about this! Actually I was confused. I didn't see that `HDFSCopyFromLocal` is using `org.apache.hadoop.fs.FileSystem` and not the Flink `FileSystem`. So it should be fine to have it in `flink-hadoop-compatability` instead. What do you think?